broker: unset SLURM_* in initial program environment

garlick opened the pull request:
Problem: SLURM_ prefixed environment variables that leak through slurm -> flux -> MPI can cause hard to diagnose problems.

Clear all SLURM_ prefixed variables in the environment of the rc scripts spawned by the broker (where rc2 == initial program).

grondo commented on src/broker/runat.c:
```diff
@@ -339,9 +337,14 @@ static int runat_command_modenv (struct runat_command *cmd,
-        for (i = 0; blocklist[i] != NULL; i++)
-            flux_cmd_unsetenv (cmd->cmd, blocklist[i]);
+        for (int i = 0; blocklist[i] != NULL; i++) {
+            if (strchr (blocklist[i], '?')
+                || strchr (blocklist[i], '*')
+                || strchr (blocklist[i], '['))
+                flux_cmd_unsetenv_glob (cmd->cmd, blocklist[i]);
+            else
+                flux_cmd_unsetenv (cmd->cmd, blocklist[i]);
```
This makes me wonder if `flux_cmd_unsetenv()` should just optionally take a glob pattern and push detection of the pattern inside that function?

Also, elsewhere we have these 3 tests abstracted into an `isa_glob()` function. If it isn't quite enough to warrant moving to libutil at this point (and it is such a small function that I don't think so either), then perhaps using `isa_glob()` here (or in `flux_cmd_unsetenv()` would at least allow the common function to easily be discovered later...

garlick replied:
Oh duh, excellent idea.  I'll do that.

grondo reviewed (commented):
LGTM! My one serious comment is whether we want to just extend `flux_cmd_unsetenv()` to optionally take a pattern, since doing the `isa_glob()` test and optionally calling `flux_cmd_unsetenv_glob()` would likely be the/a common use case...

garlick commented:
Thanks for that review.  I've implemented your suggestion of just adding glob support to `flux_cmd_unsetenv()` which makes much more sense(!) and fixed up the other stuff.  I did not create `libutil/isa_glob.c` but I did abstract that test out to an identical function so it's more clear.

grondo reviewed (approved):
LGTM!
